frontend/src/metabase/browse/components/BrowseModels.tsx

```diff
@@ -37,20 +37,17 @@ export const BrowseModels = () => {
 
   const modelsResult = useFetchModels({ model_ancestors: true });
 
-  const { allModels, doVerifiedModelsExist } = useMemo(() => {
-    const allModels =
+  const { models, doVerifiedModelsExist } = useMemo(() => {
+    const unfilteredModels =
       (modelsResult.data?.data as ModelResult[] | undefined) ?? [];
-    const doVerifiedModelsExist = allModels.some(
+    const doVerifiedModelsExist = unfilteredModels.some(
       model => model.moderated_status === "verified",
     );
-    return { allModels, doVerifiedModelsExist };
+    const models =
+      PLUGIN_COLLECTIONS.filterOutItemsFromInstanceAnalytics(unfilteredModels);
+    return { models, doVerifiedModelsExist };
   }, [modelsResult]);
 
-  const models = useMemo(
-    () => PLUGIN_COLLECTIONS.filterOutItemsFromInstanceAnalytics(allModels),
-    [allModels],
-  );
-
```

Comment:
The core change in this PR is right here. We should filter out the instance analytics models right away when we get new results from the API, and only ever use this filtered list. Previously we kept a copy of the unfiltered list and used that on line 77 below